### PR TITLE
SLING-11507 - Field injection should not inject static fields

### PR DESCRIPTION
**Pull request by rombert**
Create a ValidatingAnnotationProcessor that enforces required behaviours.

### CODE REVIEW

**Comment by rombert** [comment]
With this annotation processor the following code 

```
    @SlingObject
    private static ResourceResolver resourceResolver;
```

causes a compilation failure

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project core: Compilation failure
[ERROR] /home/robert/sources/apache/sling/tmp/project001/core/src/main/java/com/example/models/SampleRequestModel.java:[35,37] Annotation org.apache.sling.models.annotations.injectorspecific.SlingObject may not be used on static fields
```

File: src/main/java/org/apache/sling/models/annotations/apt/ValidatingAnnotationProcessor.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    processingEnv.getMessager().printMessage(
+                        Kind.ERROR,
+                        "Annotation " + annotation + " may not be used on static fields",
+                        annotatedElement
+                    );
+                } else if (annotatedElement.getKind() == ElementKind.METHOD) {
+                    processingEnv.getMessager().printMessage(
+                        Kind.ERROR,
+                        "Annotation " + annotation + " may not be used on static methods",
+                        annotatedElement
```
Comment (stefanseifert) [inline src/main/java/org/apache/sling/models/annotations/apt/ValidatingAnnotationProcessor.java:63]: i think we should include the actual method name in the log message

**Comment by Buuhuu** [comment]
@cziegeler @rombert @anchela @joerghoh can you have another look?

**Review by stefanseifert — APPROVED** [review]

**Comment by stefanseifert** [comment]
please also update the title and description of [SLING-11507](https://issues.apache.org/jira/browse/SLING-11507) with the new scope also matching final fields/final methods

**Pull request merged**